browser/pages/confirm/confirm.html

```diff
@@ -11,546 +11,22 @@
-      <div id="fusetools-panel-heading" class="panel-heading panel-normal"
-           ng-class="{'dotted-panel':checkboxModel.fusetools.hasOption('detected')}">
-        <!--ng-click="checkboxModel.jbds.changeIsCollapsed()">-->
-        <div class="checkbox-container verticalLine"> <!--ng-show="!checkboxModel.jbds.hasOption('detected')"-->
-         <input id="fusetools-checkbox" type="checkbox" ng-model="checkboxModel.fusetools.selectedOption" aria-label="Toggle ngHide" class="vallign-middle" ng-true-value="'install'" ng-false-value="'detected'">
-        </div>
-        <div class="product-container">
-          <span id="fusetools-installed-note" class="installed-note" ng-show="checkboxModel.fusetools.hasOption('detected')"><i class="fa fa-check"></i> Existing installation found</span>
-          <span id="fusetools-name" class="product-name">{{checkboxModel.fusetools.productName}}</span><span id="fusetools-version" class="product-version">{{checkboxModel.fusetools.version}}</span>
-          <div id="fusetools-description">{{checkboxModel.fusetools.productDesc}}</div>
-        </div>
-      </div>
-    </div>
+    <div ng-repeat="item in checkboxModel">
+      <component-panel item="item"></component-panel>
 
-    <!-- fuseplatform -->
-    <div id="fuseplatform-panel" class="panel panel-default panel2pxborder" ng-class="{'zero-border':checkboxModel.fuse.hasOption('detected')}">
-      <div id="fuseplatform-panel-heading" class="panel-heading panel-normal"
-           ng-class="{'dotted-panel':checkboxModel.fuseplatform.hasOption('detected')}">
-        <!--ng-click="checkboxModel.jbds.changeIsCollapsed()">-->
-        <div class="checkbox-container verticalLine"> <!--ng-show="!checkboxModel.jbds.hasOption('detected')"-->
-         <input id="fuseplatform-checkbox" type="checkbox" ng-model="checkboxModel.fuseplatform.selectedOption" aria-label="Toggle ngHide" class="vallign-middle" ng-true-value="'install'" ng-false-value="'detected'">
-        </div>
-        <div class="product-container">
-          <span id="fuseplatform-installed-note" class="installed-note" ng-show="checkboxModel.fuseplatform.hasOption('detected')"><i class="fa fa-check"></i> Existing installation found</span>
-          <span id="fuseplatform-name" class="product-name">{{checkboxModel.fuseplatform.productName}}</span><span id="fuseplatform-version" class="product-version">{{checkboxModel.fuseplatform.version}}</span>
-          <div id="fuseplatform-description">{{checkboxModel.fuseplatform.productDesc}}</div>
-        </div>
+      <div class="message-container error-message" ng-show="item.keyName === 'virtualbox' && !confCtrl.isDisabled && !checkboxModel.hyperv.isConfigured() && checkboxModel.virtualbox.virtualizationEnabled === false && (checkboxModel.cdk.selectedOption == 'install' || checkboxModel.virtualbox.selectedOption == 'install') && platform === 'win32'">
```

Comment:
Feels like this 'simple messages' have:
1. Text
2. type (error/warning)
3. `function isVisible()` or `get isVisible()`

Then we would just iterate them in directive with ng-repeat.